langchain4j-infinispan/src/main/java/dev/langchain4j/store/embedding/infinispan/LanchainItemMarshaller.java

```diff
@@ -0,0 +1,48 @@
+package dev.langchain4j.store.embedding.infinispan;
+
+import org.infinispan.protostream.MessageMarshaller;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Marshaller to read and write embeddings to Infinispan
+ */
+public class LanchainItemMarshaller implements MessageMarshaller<LangchainInfinispanItem> {
```

Comment:
We're going to remove the MessageMarshaller very soon. I advice to produce the Proto model annotating the POJO LangchainInfinispanItem.

Reply:
we can't on this use case, as the item name needs to be dynamic